src/parseTokens.ts

```diff
@@ -0,0 +1,30 @@
+// we return the payload as a string rather than JSON.parse-ing it
+// so that callers can inform TypeScript the type of their payload
+
+import { MicroFrontendToken } from './app.types';
+
+// when they JSON.parse the result of this function
+const parseJwt = (token: string): string => {
+  const base64Url = token.split('.')[1];
+  const base64 = base64Url.replace(/-/g, '+').replace(/_/g, '/');
+  const payload = decodeURIComponent(
+    window.atob(base64).replace(/(.)/g, function (m, p) {
+      const code = p.charCodeAt(0).toString(16).toUpperCase();
+      return '%' + ('00' + code).slice(-2);
+    })
+  );
+  return payload;
+};
+
+type SciGatewayToken = string | null;
+
+export const readSciGatewayToken = (): SciGatewayToken => {
```

Comment:
I presume this function is just to check we have a valid token (i.e. that we have a username) since we still use the underlying token in the Auth header

Reply:
I took this from DG and I think it's still a good idea for it to validate the token is good whenever the token is used (i.e. for requests and such). As it's stored in localStorage, technically the user could modify it in some way after all. But yeah it just checks we can still get a username out of it, which is somewhat robust but I suppose the ultimate check really is whether the backend accepts it or not. So we don't have to have the JWT parsing here if you don't think it's important I suppose

Reply:
It's fine for now, it's potentially useful if info gets put in the JWT later